packages/common/src/types/RelayTransactionRequest.ts

```diff
@@ -12,39 +12,51 @@ export interface RelayMetadata {
   maxAcceptanceBudget: PrefixedHexString
 }
 
-export interface RelayTransactionRequest {
-  relayRequest: RelayRequest
+export interface BaseRelayTransactionRequest {
   metadata: RelayMetadata
 }
 
+export interface RelayTransactionRequest extends BaseRelayTransactionRequest {
+  relayRequest: RelayRequest
+}
+
+const requestShape = {
+  from: ow.string,
+  to: ow.string,
+  data: ow.string,
+  value: ow.string,
+  nonce: ow.string,
+  gas: ow.string,
+  validUntil: ow.string
+}
+
+const metadataShape = {
+  approvalData: ow.string,
+  relayHubAddress: ow.string,
+  relayMaxNonce: ow.number,
+  signature: ow.string,
+  maxAcceptanceBudget: ow.string
+}
+
+const baseRelayDataShape = {
```

Comment:
Do you still need this separation to base- and an override?

Reply:
Not anymore